src/energyplus/ForwardTranslator/ForwardTranslateCoilUserDefined.cpp

```diff
@@ -0,0 +1,141 @@
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#include "../ForwardTranslator.hpp"
+#include "../../model/Model.hpp"
+#include "../../model/Node.hpp"
+#include "../../model/Node_Impl.hpp"
+#include "../../model/ThermalZone.hpp"
+#include "../../model/ThermalZone_Impl.hpp"
+#include "../../model/CoilUserDefined.hpp"
+#include "../../model/CoilUserDefined_Impl.hpp"
+#include "../../model/EnergyManagementSystemProgramCallingManager.hpp"
+#include "../../model/EnergyManagementSystemProgramCallingManager_Impl.hpp"
+#include "../../utilities/core/Logger.hpp"
+#include "../../utilities/core/Assert.hpp"
+#include <utilities/idd/Coil_UserDefined_FieldEnums.hxx>
+#include "../../utilities/idd/IddEnums.hpp"
+#include <utilities/idd/IddEnums.hxx>
+#include <utilities/idd/IddFactory.hxx>
+
+using namespace openstudio::model;
+
+using namespace std;
+
+namespace openstudio {
+
+namespace energyplus {
+
+  boost::optional<IdfObject> ForwardTranslator::translateCoilUserDefined(CoilUserDefined& modelObject) {
+    boost::optional<std::string> s;
+    //boost::optional<double> value;
+
+    IdfObject idfObject(IddObjectType::Coil_UserDefined);
+
+    m_idfObjects.push_back(idfObject);
+
+    // Name
+    s = modelObject.name();
+    if (s) {
+      idfObject.setName(*s);
+    }
+
+    //PlantConnectionisUsed default value
+    idfObject.setString(Coil_UserDefinedFields::PlantConnectionisUsed, "No");
+
+    // PlantConnectionInletNodeName
+    bool plantUsed = false;
+    if (boost::optional<ModelObject> mo = modelObject.waterInletModelObject()) {
+      if (boost::optional<Node> node = mo->optionalCast<Node>()) {
+        idfObject.setString(Coil_UserDefinedFields::PlantConnectionInletNodeName, node->name().get());
+        //PlantConnectionisUsed to Yes since Node is connected
+        idfObject.setString(Coil_UserDefinedFields::PlantConnectionisUsed, "Yes");
+        plantUsed = true;
+      }
+    }
+
+    // PlantConnectionOutletNodeName
+
+    if (boost::optional<ModelObject> mo = modelObject.waterOutletModelObject()) {
+      if (boost::optional<Node> node = mo->optionalCast<Node>()) {
+        idfObject.setString(Coil_UserDefinedFields::PlantConnectionOutletNodeName, node->name().get());
+      }
+    }
+    // remove plant actuators if no plant connections
+    if (!plantUsed) {
+      modelObject.plantMinimumMassFlowRateActuator().get().remove();
+      modelObject.plantMaximumMassFlowRateActuator().get().remove();
+      modelObject.plantDesignVolumeFlowRateActuator().get().remove();
+      modelObject.plantMassFlowRateActuator().get().remove();
+      modelObject.plantOutletTemperatureActuator().get().remove();
+    }
```

Comment:
This makes me wonder if these shouldn't be created/deleted at model times only when connecting/disconnecting from a plant loop. Probably leaning on the overengineering side of things though.